sympy/physics/mechanics/_system.py

```diff
@@ -798,20 +798,42 @@ def form_eoms(self, eom_method=KanesMethod):
         # KanesMethod does not accept empty iterables
         loads = self.loads if self.loads else None
         if issubclass(eom_method, KanesMethod):
+            not_allowed_kwargs = {
+                "frame", "q_ind", "u_ind", "kd_eqs", "q_dependent",
+                "u_dependent", "configuration_constraints",
+                "velocity_constraints", "forcelist", "bodies"}
+            wrong_kwargs = not_allowed_kwargs.intersection(kwargs)
+            if wrong_kwargs:
+                raise ValueError(
+                    f"The key word arguments {wrong_kwargs} are not allowed to "
+                    f"be overwritten in {eom_method.__name__}.")
             velocity_constraints = self.holonomic_constraints.diff(
                 dynamicsymbols._t).col_join(self.nonholonomic_constraints)
-            self._eom_method = KanesMethod(
-                self.frame, self.q_ind, self.u_ind, kd_eqs=self.kdes,
-                q_dependent=self.q_dep, u_dependent=self.u_dep,
-                configuration_constraints=self.holonomic_constraints,
-                velocity_constraints=velocity_constraints,
-                forcelist=loads, bodies=self.bodies,
-                explicit_kinematics=False)
+            kwargs = {"frame": self.frame, "q_ind": self.q_ind, "u_ind": self.u_ind,
+                      "kd_eqs": self.kdes, "q_dependent": self.q_dep,
+                      "u_dependent": self.u_dep,
+                      "configuration_constraints": self.holonomic_constraints,
+                      "velocity_constraints": velocity_constraints,
+                      "forcelist": loads, "bodies": self.bodies,
+                      "explicit_kinematics": False, **kwargs}
+            self._eom_method = KanesMethod(**kwargs)
         elif issubclass(eom_method, LagrangesMethod):
-            self._eom_method = eom_method(
-                Lagrangian(self.frame, *self.bodies), self.q, loads,
-                self.bodies, self.frame, self.holonomic_constraints,
-                self.nonholonomic_constraints)
+            not_allowed_kwargs = {
+                "frame", "qs", "forcelist", "bodies", "hol_coneqs",
+                "nonhol_coneqs", "Lagrangian"}
+            wrong_kwargs = not_allowed_kwargs.intersection(kwargs)
+            if wrong_kwargs:
+                raise ValueError(
+                    f"The key word arguments {wrong_kwargs} are not allowed to "
```

Comment:
```suggestion
                    f"The keyword arguments {wrong_kwargs} are not allowed to "
```

Reply:
Addressed in 1c4af023695dcaf60f2edfb87350183363e84b42